timelock-server/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosTimeLockServer.java

```diff
@@ -176,49 +182,80 @@ private LeaderConfig getLeaderConfig(TimeLockServerConfiguration configuration)
                 timestampService);
     }
 
-    private AsyncTimelockService createAsyncTimelockService(String client) {
-        Supplier<ManagedTimestampService> rawTimestampServiceSupplier = createRawPaxosBackedTimestampServiceSupplier(
-                client);
-
-        Supplier<AsyncTimelockService> rawAsyncTimelockServiceSupplier = () -> {
-            ScheduledExecutorService reaperExecutor = Executors.newSingleThreadScheduledExecutor(
-                    new ThreadFactoryBuilder()
-                            .setNameFormat("async-lock-reaper-" + client + "-%d")
-                            .setDaemon(true)
-                            .build());
-            ScheduledExecutorService timeoutExecutor = Executors.newSingleThreadScheduledExecutor(
-                    new ThreadFactoryBuilder()
-                            .setNameFormat("async-lock-timeouts-" + client + "-%d")
-                            .setDaemon(true)
-                            .build());
-            return new AsyncTimelockServiceImpl(
-                    AsyncLockService.createDefault(reaperExecutor, timeoutExecutor),
-                    rawTimestampServiceSupplier.get());
-        };
-
+    private AsyncTimelockService createAsyncTimelockService(
+            String client,
+            Supplier<ManagedTimestampService> rawTimestampServiceSupplier) {
         return AwaitingLeadershipProxy.newProxyInstance(
                 AsyncTimelockService.class,
-                rawAsyncTimelockServiceSupplier,
+                () -> createRawAsyncTimelockService(client, rawTimestampServiceSupplier),
                 leaderElectionService);
     }
 
+    private AsyncTimelockService createRawAsyncTimelockService(
+            String client,
+            Supplier<ManagedTimestampService> timestampServiceSupplier) {
+        ScheduledExecutorService reaperExecutor = Executors.newSingleThreadScheduledExecutor(
+                new ThreadFactoryBuilder()
+                        .setNameFormat("async-lock-reaper-" + client + "-%d")
+                        .setDaemon(true)
+                        .build());
+        ScheduledExecutorService timeoutExecutor = Executors.newSingleThreadScheduledExecutor(
+                new ThreadFactoryBuilder()
+                        .setNameFormat("async-lock-timeouts-" + client + "-%d")
+                        .setDaemon(true)
+                        .build());
+        return new AsyncTimelockServiceImpl(
+                AsyncLockService.createDefault(reaperExecutor, timeoutExecutor),
+                timestampServiceSupplier.get());
+    }
+
+    private TimeLockServices createLegacyTimeLockServices(
+            Supplier<ManagedTimestampService> rawTimestampServiceSupplier,
+            RemoteLockService lockService) {
+        TimelockService timelockService = new LockTokenConvertingTimelockService(
+                new LegacyTimelockService(rawTimestampServiceSupplier.get(),
```

Comment:
Ah yes, of course.
As discussed offline this did fix the issue we were seeing with the server failing to start as well (because it would try and create this too early)